caravel/viz.py

```diff
@@ -147,14 +147,36 @@ def get_df(self, query_obj=None):
         self.results = None
 
         # The datasource here can be different backend but the interface is common
+        timestamp_format = None
+        if self.datasource.type == 'table':
+            dttm_cols = self.datasource.dttm_cols
+            col_name = query_obj['granularity']
+            if col_name not in dttm_cols:
+                col_name = self.datasource.main_dttm_col
+            columns = self.datasource.table_columns
+            for col in columns:
+                if col_name == col.column_name:
+                    timestamp_format = col.python_date_format
+                    break
+
         self.results = self.datasource.query(**query_obj)
         self.query = self.results.query
         df = self.results.df
+        # Transform the timestamp we received from database to pandas supported
+        # datetime format. If no python_date_format is specified, the pattern will
+        # be considered as the default ISO date format
+        # If the datetime format is unix, the parse will use the corresponding
+        # parsing logic.
         if df is None or df.empty:
             raise Exception("No data, review your incantations!")
         else:
             if 'timestamp' in df.columns:
-                df.timestamp = pd.to_datetime(df.timestamp, utc=False)
+                if timestamp_format == "epoch":
+                    df.timestamp = pd.to_datetime(
+                        df.timestamp, utc=False, unit="s")
+                else:
+                    df.timestamp = pd.to_datetime(
+                        df.timestamp, utc=False, format=timestamp_format)
```

Comment:
Does this work for databases that do have a DATETIME or DATE type? Seems like it would only work for strings.